src/steps/ms-defender/device/index.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Entity,
+  IntegrationStepExecutionContext,
+  Step,
+} from '@jupiterone/integration-sdk-core';
+
+import { IntegrationConfig, IntegrationStepContext } from '../../../config';
+import { DefenderClient } from '../clients/defenderClient';
+
+import {
+  DATA_ACCOUNT_ENTITY,
+  entities,
+  MappedRelationships,
+  relationships,
+  steps,
+  TargetEntities,
+} from '../constants';
+import {
+  createMachineEntity,
+  createMachinesDeviceRelationship,
+  createAccountMachineRelationship,
+} from './converters';
+
+import { steps as activeDirectorySteps } from '../../active-directory/constants';
+
+export async function fetchMachines(
+  executionContext: IntegrationStepContext,
+): Promise<void> {
+  const { logger, instance, jobState } = executionContext;
+  const graphClient = new DefenderClient(logger, instance.config);
+
+  const accountEntity = await jobState.getData<Entity>(DATA_ACCOUNT_ENTITY);
+  if (!accountEntity) {
+    logger.warn('Error fetching machine data: accountEntity does not exist');
+    return;
+  }
+  await graphClient.iterateMachines(async (machine) => {
+    const machineEntity = createMachineEntity(machine);
+    await jobState.addEntity(machineEntity);
+    await jobState.addRelationship(
+      createAccountMachineRelationship(accountEntity, machineEntity),
+    );
+  });
+}
+
+export async function buildMachineManagesDevicesRelationships(
+  executionContext: IntegrationStepContext,
+): Promise<void> {
+  const { jobState } = executionContext;
+
+  await jobState.iterateEntities(
+    { _type: entities.MACHINE._type },
+    async (machine) => {
+      await jobState.addRelationship(createMachinesDeviceRelationship(machine));
+    },
+  );
+}
+
+export const deviceSteps: Step<
+  IntegrationStepExecutionContext<IntegrationConfig>
+>[] = [
+  {
+    id: steps.FETCH_MACHINE,
+    name: 'Microsoft Defender Machines',
```

Comment:
Admittedly, we've been inconsistent in the name of steps. I took a look at a few integrations and https://github.com/JupiterOne/sdk/blob/main/docs/integrations/step-patterns.md, and there is a pattern emerging.

```suggestion
    name: 'Fetch Machines',
```